### PR TITLE
add more details needed status

### PR DESCRIPTION
**Pull request by Reception123**
partially implementing T11735 (who's scope slightly changed following WC discussion)

### CODE REVIEW

File: includes/CreateWikiNotificationsManager.php
Comment (OAuthority) [inline includes/CreateWikiNotificationsManager.php:1]: Typo

File: includes/Notifications/EchoRequestMoreDetailsPresentationModel.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Miraheze\CreateWiki\Notifications;
+
+use EchoDiscussionParser;
+use EchoEventPresentationModel;
+use RawMessage;
+
+class EchoRequestDeclinedPresentationModel extends EchoEventPresentationModel {
```
Comment (redbluegreenhat) [inline includes/Notifications/EchoRequestMoreDetailsPresentationModel.php:9]: i18 for the new messages needs to be introduced in en.json

File: includes/RequestWiki/SpecialRequestWikiQueue.php [hunk 2]
```diff
@@ -60,7 +60,8 @@ private function doPagerStuff() {
 					'Unreviewed' => 'inreview',
 					'Approved' => 'approved',
 					'Declined' => 'declined',
-					'On hold' => 'onhold',
+					'On hold (further review)' => 'onhold',
+					'Needs more details' => 'moredetails',
```
Comment (redbluegreenhat) [inline includes/RequestWiki/SpecialRequestWikiQueue.php:64]: This option needs to be introduced to RequestWikiRequestViewer, in the `submission-action` key of the form descriptor (https://github.com/miraheze/CreateWiki/blob/master/includes/RequestWiki/RequestWikiRequestViewer.php#L244).

**Review by redbluegreenhat — CHANGES_REQUESTED** [review]

File: includes/Notifications/EchoRequestMoreDetailsPresentationModel.php (hunk 1 above)
Comment (redbluegreenhat) [reply includes/Notifications/EchoRequestMoreDetailsPresentationModel.php:9]: Also, change the class name to `EchoRequestMoreDetailsPresentationModel.php`

Comment (OAuthority) [reply includes/Notifications/EchoRequestMoreDetailsPresentationModel.php:9]: Class is already declared elsewhere, needs to be renamed.

File: includes/RequestWiki/SpecialRequestWikiQueue.php (hunk 2 above)
Comment (redbluegreenhat) [reply includes/RequestWiki/SpecialRequestWikiQueue.php:64]: Will also need to add some logic in `submitForm` (https://github.com/miraheze/CreateWiki/blob/master/includes/RequestWiki/RequestWikiRequestViewer.php#L342)

Comment (redbluegreenhat) [reply includes/RequestWiki/SpecialRequestWikiQueue.php:64]: and in the `WikiRequest` class.

**Comment by redbluegreenhat** [comment]
For the logs, see https://github.com/miraheze/CreateWiki/blob/da4bbf1152b1b3e30f0148d79facd522120dd0ed/extension.json#L66C1-L73C4

Just add the log there and keep the default LogFormatter as the formatter.

File: includes/CreateWikiNotificationsManager.php [hunk 3]
```diff
@@ -91,6 +91,7 @@ private function getEchoTypes(): array {
 		return [
 			'request-comment',
 			'request-declined',
+			'reques-moredetails',
```
Comment (redbluegreenhat) [inline includes/CreateWikiNotificationsManager.php]: typo

**Review by redbluegreenhat — CHANGES_REQUESTED** [review]

File: includes/Notifications/EchoRequestMoreDetailsPresentationModel.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+use EchoEventPresentationModel;
+use RawMessage;
+
+class EchoRequestMoreDetailsPresentationModel extends EchoEventPresentationModel {
+	public function getIconType() {
+		return 'request-moredetails';
+	}
+
+	public function getHeaderMessage() {
+		return $this->msg( 'notification-header-request-moredetails' );
```
Comment (redbluegreenhat) [inline includes/Notifications/EchoRequestMoreDetailsPresentationModel.php:15]: this message is missing from en.json

**Review by redbluegreenhat — CHANGES_REQUESTED** [review]

File: includes/RequestWiki/WikiRequest.php [hunk 5]
```diff
@@ -246,6 +246,24 @@ public function onhold( string $reason, User $user ) {
...
+		$notifyUsers = $this->involvedUsers;
+		unset(
+			$notifyUsers[$this->requester->getId()],
+			$notifyUsers[$user->getId()]
+		);
+
+		if ( $notifyUsers ) {
+			$this->sendNotification( $reason, $notifyUsers );
+		}
+		$this->log( $user, 'moredetails' );
```
Comment (redbluegreenhat) [inline includes/RequestWiki/WikiRequest.php:264]:
```suggestion
		$this->log( $user, 'requestmoredetails' );
```

**Review by redbluegreenhat — CHANGES_REQUESTED** [review]

**Review by redbluegreenhat — APPROVED** [review]

**Pull request merged**